lib/ukrust/Config.uk

```diff
@@ -0,0 +1,9 @@
+config LIBUKRUST
+	bool "ukrust: Rust support for Unikraft's internals"
+	depends on ARCH_X86_64
+	default n
```

Comment:
At least this line needs to be added because libc headers are used in the glue code (like stddef)
```
select LIBNOLIBC if !HAVE_LIBC
```
Better would be even to compile with unikraft headers only - then this dependency can go away

Reply:
We have Unikraft-only types, like `__64`, `__sz`, `__uptr`

Reply:
It would be great to treat dependency to `LIBUKALLOC` optional. Either return `ENOMEM` on library absence or exclude it from the build if possible.

Reply:
I don't think it makes sense to use this without LIBUKALLOC since you won't be able to use neither the core nor the alloc or bindings crate. In this case you'd have to use barebones rust.(what we have right now) I've removed the stddef include, we no longer need any libc.

Reply:
Wait, why you need LIBUKALLOC for all that stuff?
Rust and core should work fine without an allocator. Everything that needs an allocator is in `alloc` afaik. Also not sure why that would have an impact on bindgen if we `#ifdef` the header inclusion

Reply:
Oh, you're right, #ifdef should do the trick for bindgen, let me fix it quickly.